### PR TITLE
[FOSSA] Add Vulnerability and License Dependency Details View, and Error Handling

### PR DESCRIPTION
**Pull request by angieyeh**
This PR handles NR-119857 which is for adding modal details view. Also did clean up for imports and exports, and added error handling.

Vuln modal
Screenshot 2023-07-26 at 5 13 02 PM
Lic Dep modal
Screenshot 2023-07-26 at 5 13 15 PM
Fetch vuln and lic dep error
Screenshot 2023-07-26 at 5 11 20 PM
Only vuln error
Screenshot 2023-07-27 at 11 28 02 AM
Only lic dep error
Screenshot 2023-07-27 at 11 29 06 AM
No repo match error
Screenshot 2023-07-27 at 12 13 28 PM

### CODE REVIEW

**Comment by angieyeh** [comment]
Leaving it as draft pr until https://github.com/TeamCodeStream/codestream/pull/1632 gets merged.

File: shared/agent/src/providers/fossa.ts [hunk 1]
```diff
@@ -167,16 +164,24 @@ export class FossaProvider extends ThirdPartyCodeAnalyzerProviderBase<CSFossaPro
...
+			}
+			const projects: FossaProject[] = await this._getProjects();
+			const project = await this._matchRepoToFossaProject(currentRepo, projects, request.repoId);
+			if (!project) {
+				return { isRepoMatch: false };
+			}
+			return { isRepoMatch: true };
+		} catch (error) {
+			Logger.error(error);
+			if (error.message === "Unauthorized") {
```
Comment (eric-j) [inline shared/agent/src/providers/fossa.ts:180]: Would be safer to use an error code here if available.  


If none is available from fossa error response, then maybe standardize the string to lower case before doing equals check.

**Review by eric-j — APPROVED** [review]
one small thing otherwise :+1:

File: shared/agent/src/providers/fossa.ts (hunk 1 above)
Comment (angieyeh) [reply shared/agent/src/providers/fossa.ts:180]: when i make a request in postman there is a property called httpStatusCode, but when i tried to access that property on the error object it didn't exist 🤔 hence the equality check. standardizing sounds good and would make it safer thanks!

Comment (angieyeh) [reply shared/agent/src/providers/fossa.ts]: after reading some more about JS errors, when an error is caught, the error variable that gets passed to the catch is the built in JS error object which contains the name, message and stack properties. it seems like behind the scenes, JS is parsing the error response we get back from fossa. is there anyway to access the httpStatusCode, or is the built in JS error object the best we can get?

**Pull request merged**